### PR TITLE
[kube-prometheus-stack] Separate config reloader limit and request

### PR DESCRIPTION
**Pull request by ctrox**
#### What this PR does / why we need it:

Previously it was not possible to set different values for requests/limits of the config reloader. This simply separates `configReloaderCpu` and `configReloaderMemory` into two each for specifying both the limit and request.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 18.1.0
+version: 18.2.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:21]: As this changes the current value i would consider this a breaking change. Therefor major version change needed.

```suggestion
version: 19.0.0
```

File: charts/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -1608,13 +1608,15 @@ prometheusOperator:
     tag: v0.50.0
     sha: ""
 
-  ## Set the prometheus config reloader side-car CPU limit
+  ## Set the prometheus config reloader side-car CPU resources
```
Comment (monotek) [inline charts/kube-prometheus-stack/values.yaml:1611]: As we break this already i would use the chance to do it like this:

```
  ## Prometheus-config-reloader
  ##
  prometheusConfigReloader:
    # image to use for config and rule reloading
    image:
      repository: quay.io/prometheus-operator/prometheus-config-reloader
      tag: v0.50.0
      sha: ""

    # resource config for prometheusConfigReloader
    resources:
      requests:
        cpu: 100m
        memory: 50Mi
      limits:
        cpu: 100m
        memory: 50Mi
```

deployments.yaml needs to be adjusted acordingly.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (ctrox) [reply charts/kube-prometheus-stack/Chart.yaml]: Makes sense, changed it.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 19.2.2
+version: 20.0.0
```
Comment (gkarthiks) [inline charts/kube-prometheus-stack/Chart.yaml:21]: apologies for the delayed response. can you please fix the conflicts, I will approve this PR

**Review by gkarthiks — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 3 above)
Comment (ctrox) [reply charts/kube-prometheus-stack/Chart.yaml]: No worries, I also completely forgot about this PR. Just rebased it and bumped the version again.

**Review by monotek — APPROVED** [review]

**Comment by JoKeRGreGre** [comment]
Is it possible to externalize the configuration of the config-loader for the Prometheus StatefulSet as well?

In case the initContainer doesn't have enough resources, Prometheus can't start, and it's not possible to change this configuration in the StatefulSet because it's overwritten by the Prometheus object.

For those who also encounter this issue, one solution would be to scale down the Prometheus to 0, clone the StatefulSet by removing the labels of the operator: operator.prometheus.io/*, and then modify the resources of the initContainers.

**Pull request merged**